relayer/pkg/monitoring/config_chain.go

```diff
@@ -0,0 +1,112 @@
+package monitoring
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"time"
+
+	relayMonitoring "github.com/smartcontractkit/chainlink-relay/pkg/monitoring"
+)
+
+type StarknetConfig struct {
+	RPCEndpoint      string
+	NetworkName      string
+	NetworkID        string
+	ChainID          string
+	ReadTimeout      time.Duration
+	PollInterval     time.Duration
+	LinkTokenAddress string
+}
+
+var _ relayMonitoring.ChainConfig = StarknetConfig{}
+
+func (s StarknetConfig) GetRPCEndpoint() string         { return s.RPCEndpoint }
```

Comment:
Do we need getters for public properties?

Reply:
Maybe then convert them all into private properties with public getters?
We have mixed coding styles across the stack (private properties with public getters vs public properties). I wonder what the idiomatic go vision on that. I'd bet on public properties since it's less code.
Nit for now, just to keep in mind for future extensions.

Reply:
A bit more context here. This `StarknetConfig` struct contains chain-level configuration for the monitor. In chainlink-solana there is a SolanaConfig. These two share some fields but also have chain-specific fields. 

The fields that are shared are also needed/required by the [core on-chain monitoring library](https://github.com/smartcontractkit/chainlink-relay/tree/main/pkg/monitoring), that's why they are encoded as methods of an interface. 

I agree with you that making these fields public with also public getters is strange!

Reply:
> Maybe then convert them all into private properties with public getters?

Reply:
Done in https://github.com/smartcontractkit/chainlink-starknet/pull/103/commits/0a5ebae47b94e10bde3b97b9e3686d315764d139